src/utils/schema/widgetPropertyUpdaters.ts

```diff
@@ -1,6 +1,136 @@
 import { VernaContextProps } from '../../providers/VernaProvider';
 import VernaJSONSchemaType from '../../types/rjsf';
 import type { Properties } from './updateWidgetProperties';
+import { RJSF_ID_SEPARATOR } from '../../settings';
+import { Maybe } from '../../types/utils';
+
+/**
+ * Generate the translation key based on those parameters
+ * The key is built on the id of the widget (cd: "root_section_widget") with the name of
+ * the property translated
+ *
+ * @param widgetPath is the decomposed id of the widget
+ * @param propertyKey is the name of the property we are generating a key for
+ * @param selector is the name of the selected section if there is one
+ */
+function getTranslationKey(widgetPath: string[], propertyKey: string, selector: Maybe<string>) {
+  if (selector) {
+    const translationKeyParts = [widgetPath.shift(), selector, ...widgetPath, propertyKey];
+    return translationKeyParts.join(RJSF_ID_SEPARATOR);
+  } else {
+    return [...widgetPath, propertyKey].join(RJSF_ID_SEPARATOR);
+  }
+}
+
+/**
+ * Destroy old translation values of the propertyKey element
+ *
+ * @param propertyKey The key of the property which translations have to be removed
+ * @param verna Verna context
+ * @param translationKey the key of translation used for this element
+ * @param locale Current locale
+ */
+function removePropertyTranslations(
+  propertyKey: keyof VernaJSONSchemaType,
+  verna: VernaContextProps,
+  translationKey: string,
+  locale: string,
+) {
+  if (locale) {
+    translationKey.replace(`_${propertyKey}`, '');
+    const oldKeys = Object.keys(verna.schemaTranslations[locale]).filter((key) =>
+      key.includes(translationKey),
+    );
+    oldKeys.forEach((key) => {
+      delete verna.schemaTranslations[locale][key];
+    });
+  }
+}
+
+function updateItemsTranslations(widget: VernaJSONSchemaType, translationKeys: string[]) {
+  if (Array.isArray(widget.items)) {
+    // TODO: add other properties management & matching between those and translation keys
+    widget.items = translationKeys.map((key: string) => {
+      return {
+        enum: [key],
+      };
+    });
+  } else {
+    widget.items!.enum = translationKeys;
+  }
+}
+
+/**
+ * Update the translations for an enum
+ * Each value has a unique translation key
+ *
+ * @param propertyKey The key of the property to modify
+ * @param widget The schema part representing the current widget
+ * @param verna The verna context, used to update the rendered schema and the translations
+ * @param values The value is a list of strings that we set to this key
+ * @param translationKey the corresponding key used in the translation object
+ * @param locale The current locale loaded
+ */
+function updateEnumTranslations(
+  propertyKey: keyof VernaJSONSchemaType,
+  widget: VernaJSONSchemaType,
+  verna: VernaContextProps,
+  values: string[],
+  translationKey: string,
+  locale: string,
+) {
+  if (locale) {
```

Comment:
No need to check if locale is undefined as this is a string in all case.

```suggestion
```

Reply:
Ah yes forgot to change it